src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +43,30 @@
      * Андрей - 600
      * София - 900</p>
      */
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d.MM.yyyy");
+        LocalDate dateStart = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateEnd = LocalDate.parse(dateTo, formatter);
+        if (dateStart.compareTo(dateEnd) > 0) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+        StringBuilder sb = new StringBuilder("Отчёт за период ");
+        sb.append(dateFrom).append(" - ").append(dateTo).append("\n");
+        for (int j = 0; j < names.length; j++) {
+            int salary = 0;
+            for (int i = 0; i < data.length; i++) {
+                LocalDate ld = LocalDate.parse(data[i].substring(0, 10), formatter);
+                String[] dataArray = data[i].split(" "); //26.04.2019 София 9 100
+                if ((ld.compareTo(dateStart) >= 0
+                        & ld.compareTo(dateEnd) <= 0)
```

Comment:
isBefore or isAfter, please :)

Reply:
I am not agree. What if  ld=dateStart?  In this case I need to have true but when I use isAfter or isBefore I have false